aiida/cmdline/params/types/plugin.py

```diff
@@ -68,49 +67,52 @@ def __init__(self, group=None, load=False, *args, **kwargs):
         is not specified use the tuple of all recognized entry point groups.
         """
         # pylint: disable=keyword-arg-before-vararg
+        self.load = load
+        self._input_group = group
+
+        super().__init__(*args, **kwargs)
+
+    def _get_valid_groups(self):
```

Comment:
Why have this method _and_ the `groups` property? AFAICT `_get_valid_groups` is only called by `groups` so why not just make this the cached property?

Reply:
I meant this essentially just to have the code cleaner, i.e. having the implementation details  in a separate function, to keep the property itself simple. But can change it if you think it would be less confusing.

Reply:
If it were used elsewhere I would agree that it would help, but if `groups` then solely just calls this method and it is the only caller, I don't really see the point

Reply:
Okay, done.